taker/src/lib.rs

```diff
@@ -0,0 +1,539 @@
+    #[clap(long)]
+    maker: Option<String>,
+
+    /// The public key of the maker as a 32 byte hex string.
+    ///
+    /// If not specified it defaults to the itchysats maker-id for mainnet or testnet.
+    #[clap(long, parse(try_from_str = parse_x25519_pubkey))]
+    maker_id: Option<x25519_dalek::PublicKey>,
+
+    /// Maker's peer id, required for establishing libp2p encrypted connection.
+    ///
+    /// If not specified it defaults to the itchysats maker-peer-id for mainnet or testnet.
+    #[clap(long)]
+    maker_peer_id: Option<PeerId>,
+
+    /// The IP address to listen on for the HTTP API.
+    #[clap(long, default_value = "127.0.0.1:8000")]
+    http_address: SocketAddr,
+
+    /// Where to permanently store data, defaults to the current working directory.
+    #[clap(long)]
+    data_dir: Option<PathBuf>,
+
+    /// If enabled logs will be in json format
+    #[clap(short, long)]
+    json: bool,
+
+    /// If enabled, logs in json format will contain a list of all ancestor spans of log events.
+    /// This **only** has an effect when `json` is also enabled.
+    #[clap(long)]
+    pub json_span_list: bool,
+
+    /// If enabled, traces will be exported to the OTEL collector
+    #[clap(long)]
+    instrumentation: bool,
+
+    /// If enabled, tokio runtime can be locally debugged with tokio_console
+    #[clap(long)]
+    pub tokio_console: bool,
+
+    /// If enabled, libp2p ping and offer broadcast spans will be included in the traces exported
+    /// by the application.
+    #[clap(long)]
+    pub verbose_spans: bool,
+
+    /// OTEL collector endpoint address
+    ///
+    /// If not specified it defaults to the local collector endpoint.
+    #[clap(long, default_value = LOCAL_COLLECTOR_ENDPOINT )]
+    collector_endpoint: String,
+
+    /// If enabled, browser UI is not automatically launched at startup.
+    #[clap(long)]
+    pub headless: bool,
+
+    /// Service name for OTEL.
+    ///
+    /// If not specified it defaults to the binary name.
+    #[clap(long, default_value = "taker")]
+    service_name: String,
+
+    /// Configure the log level, e.g.: one of Error, Warn, Info, Debug, Trace
+    #[clap(short, long, default_value = "Debug")]
+    log_level: LevelFilter,
+
+    /// Password for the web interface.
+    ///
+    /// If not provided, will be derived from the seed.
+    #[clap(long)]
+    password: Option<Password>,
+
+    #[clap(subcommand)]
+    network: Option<Network>,
+
+    #[clap(short, long, parse(try_from_str = parse_app_seed))]
+    app_seed: Option<[u8; 32]>,
+
+    /// If provided will be used for internal wallet instead of a random key or app_seed. The
+    /// keys will be derived according to Bip84.
+    #[clap(short, long)]
+    pub wallet_xprv: Option<ExtendedPrivKey>,
+}
+
+impl Opts {
+    pub fn read() -> Opts {
+        Opts::parse()
+    }
+
+    pub fn new(network: String, password: String, data_dir: String) -> Opts {
```

Comment:
well today its only used for electron, but basically this is a way of initializing the opts without the use of stdin. This is needed when we are using the taker as library, e.g. for electron, but it would be the same for android, iOS, etc. Hence I wouldn't call it new_electron - if we need to make it concrete, we could call it new_neon - as this is used by the neon-binding wrapper - which will then produce a nodejs module that is used by electron.